src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -442,36 +442,39 @@ object Parser {
       (idnDef <~ "=") ~ typ ^^ { case left ~ right => PTypeAlias(right, left)}
 
     lazy val functionDecl: Parser[PFunctionDecl] =
-      functionSpec ~ ("func" ~> idnDef) ~ signature ~ specOnlyParser(blockWithBodyParameterInfo) ^^ {
-        case spec ~ name ~ sig ~ body =>
-          PFunctionDecl(name, sig._1, sig._2, spec, body)
-      }
+      for {
+        spec <- functionSpec
+        name <- "func" ~> idnDef
+        sig  <- signature
+        body <- if (spec.isTrusted) nestedCurlyBracketsConsumer else specOnlyParser(blockWithBodyParameterInfo)
+      } yield PFunctionDecl(name, sig._1, sig._2, spec, body).at(spec)
```

Comment:
Is it necessary to  call `.at(spec)` here? I would expect that Kiama sets the positional information itself but I'm not sure in the context of `for ... yield`. Letting Kiama do it would have the advantage that the end position is presumably after the body. As you use `spec` right now, the function's position only spans pre- & postconditions

Reply:
You are correct, I replaced `.at(...)` with `.range(...)`. 

I don't really know if it makes more sense to let Kiama handle the position, since, with a solution where the spec gets parsed before the body, Kiama wouldn't include the spec in the function position. 

The previous solution, where I let Kiama set the position correctly(see `src/main/scala/viper/gobra/frontend/Parser.scala` in [62cdff1](https://github.com/viperproject/gobra/pull/378/commits/62cdff16b205bda2713f456a2d313215f5d25845)), is, as joao said a little bit repetitive and unelegant.